### PR TITLE
feat: WebAssembly filter chains

### PR DESCRIPTION
**Pull request by hishamhm**
This PR introduces the FilterChains entity, a new entity released in Kong Gateway 3.4 (OSS and Enterprise) to manage WebAssembly filters ("filters" are the WebAssembly analog to Lua plugins — we call them filters because it's the already-established terminology used by the proxy-wasm API we implemented, which is also implemented by other WebAssembly-enabled gateways such as Envoy).

---

KAG-4005

### CODE REVIEW

**Comment by hishamhm** [comment]
Regarding the failing test: we probably need `KONG_WASM=on` set on the 3.4/nightly tests for the filter chain operations to work, and probably `KONG_WASM_FILTERS_PATH=filters/` as well, containing a dummy `filters/example-filter.wasm`

**Comment by flrgh** [comment]
CI problems...

The nightly tests were using an old tag (`master-alpine`), so those tests were broken for filter chains. I updated this to `master`. _However_, now some other tests are failing now that they are being executed against a real nightly image:

```
=== RUN   TestFillPluginDefaults/no_config_no_protocols
    plugin_service_test.go:671:   &kong.Plugin{
          	... // 6 identical fields
          	Consumer:      nil,
          	ConsumerGroup: nil,
          	Config: kong.Configuration{
          		"anonymous":        nil,
          		"hide_credentials": bool(false),
        - 		"realm":            string("service"),
          	},
          	Enabled: &true,
          	RunOn:   &"test",
          	... // 3 identical fields
          }
```

**Comment by flrgh** [comment]
Lint fixes are in #421

File: kong/plugin_service_test.go [hunk 1]
```diff
@@ -553,6 +551,9 @@ func TestFillPluginDefaults(T *testing.T) {
 	SkipWhenEnterprise(T)
 	RunWhenKong(T, ">=2.3.0")
 
+	// FIXME: this test needs to be more version-aware due to new fields added 3.7
```
Comment (flrgh) [inline kong/plugin_service_test.go:554]: https://github.com/Kong/go-kong/pull/362#issuecomment-2010158479

**Comment by flrgh** [comment]
I had to fix a host of testing problems in order to get CI working against Kong nightly/master, so there are a bunch of additional commits in this PR. I can move them to another PR if desired.

**Comment by flrgh** [comment]
@hishamhm who is a good person to tag for review on this?

**Review by rainest — APPROVED** [review]
The consumer groups stuff should have probably gone in separately but otherwise the new entity looks fine.

**Comment by pmalek** [comment]
https://github.com/Kong/go-kong/pull/362#issuecomment-2010158479 is being addressed in https://github.com/Kong/go-kong/pull/432

**Pull request merged**